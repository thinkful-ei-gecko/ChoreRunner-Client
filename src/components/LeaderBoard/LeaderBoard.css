/* this is in the wrong spot */

.leaderboard_container {
  background: rgba(0, 0, 0, 0.7);
  border-radius: 10px;
  width: 60%;
  min-width: 40%;
  margin: 20px auto;
}

.leader_board {
  font-family: 'Press Start 2P', cursive;
  color: white;
  text-align: center;
  padding: 10px;
  border: 2px solid white;
}

.headerRow {
  width: auto;
  display: flex;
  padding: 5px 0;
  border-bottom: 1px solid white;
}

.header_rank {
  width: 20%;
  text-align: left;
  padding: 0 5px;
}

.header_name {
  width: 50%;
  text-align: center;
  padding: 0 5px;
}

.header_score {
  width: 20%;
  text-align: right;
  padding-right: 20px;
}

.score_display {
  width: auto;
  padding-left: 25px;
  margin: 10px auto;
}

.rankings {
  width: 80%;
  margin: 0 auto;
}

.rankings li:nth-child(-n+3):before {
  content: '';
  display: inline-block;
  height: 25px;
  width: 25px;
  background-size: contain;
  background-repeat: no-repeat;
  margin-left: -65px;
  background-image: url('../../images/FirstCup.png');
}


.rankings li:nth-child(1):before {
  background-image: url('../../images/FirstCup.png');
}

.rankings li:nth-child(2):before {
  background-image: url('../../images/SecondCup.png');
}

.rankings li:nth-child(3):before {
  background-image: url('../../images/ThirdCup.png');
}

.rankings li:nth-child(-n+3) .score {
  display: block;
}


.name_col {
  width: 70%;
  text-align: center;
  margin: 0 auto;
  float: left;
  word-wrap: break-word;
}

.score_col {
  width: 20%;
  float: right;
}
