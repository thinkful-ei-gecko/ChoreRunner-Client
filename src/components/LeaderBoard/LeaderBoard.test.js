import React from 'react';
import ReactDOM from 'react-dom';
import LeaderBoard from './LeaderBoard';
import {MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe.skip('LeaderBoard component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
  
    ReactDOM.render(<MemoryRouter><LeaderBoard /></MemoryRouter>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter><LeaderBoard /></MemoryRouter>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  })

  it('renders empty given no tabs', () => {
    const wrapper = shallow(<LeaderBoard />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('reads the components State and lifecycle method', () => {
    const wrapper = shallow(<LeaderBoard />)
    const compInstance = wrapper.instance();
    //Access lifecycle method
    compInstance.componentDidMount()
    expect(wrapper.state('members')).toEqual([])
  })
})
