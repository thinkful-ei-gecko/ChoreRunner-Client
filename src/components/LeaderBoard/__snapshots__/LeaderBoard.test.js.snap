// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderBoard component testing renders UI as expected 1`] = `
<section
  className="leader_board"
>
  <h3>
    Leaderboard
  </h3>
  <div
    className="headerRow"
  >
    <div
      className="header_rank"
    >
      <span>
        Rank
      </span>
    </div>
    <div
      className="header_name"
    >
      <span>
        Member Name
      </span>
    </div>
    <div
      className="header_score"
    >
      <span>
        Score
      </span>
    </div>
  </div>
  <ol
    className="rankings"
  />
</section>
`;

exports[`LeaderBoard component testing renders empty given no tabs 1`] = `
<section
  className="leader_board"
>
  <h3>
    Leaderboard
  </h3>
  <div
    className="headerRow"
  >
    <div
      className="header_rank"
    >
      <span>
        Rank
      </span>
    </div>
    <div
      className="header_name"
    >
      <span>
        Member Name
      </span>
    </div>
    <div
      className="header_score"
    >
      <span>
        Score
      </span>
    </div>
  </div>
  <ol
    className="rankings"
  />
</section>
`;
